app/views/partials/view-claim/claim.html

```diff
@@ -36,6 +40,9 @@
               <span class="text-pending">To be posted later</span>
               <a href="/claim/file-upload/{{ Claim['Reference'] }}/{{ Claim['ClaimId'] }}/RECEIPT?eligibilityId={{ Claim['EligibilityId'] }}&claimExpenseId={{ expense.ClaimExpenseId }}&claimDocumentId={{ expense.ClaimDocumentId }}" class="button grey pull-right">Add</a>
 
+            {% elif expense.ExpenseType == 'train' and Claim.IsAdvanceClaim %}
```

Comment:
Instead of IsAdvanceClaim should it look to see if it is after the date of journey to know whether to say 'Receipt to be uploaded after visit' or 'Receipt required'